libcsight/include/csight/Utils.hpp

```diff
@@ -34,5 +34,10 @@ namespace csight::utils {
     return accumulator;
   }
 
+  template <typename T>
+  bool waitingInterval(std::future<T>* toWait, int interval_ms) {
```

Comment:
We'll want to add `#include <future>` to the top of the file.  At the moment the build pipeline isn't able to compile the code since it's missing the header.

Reply:
Oh, this worked on my machine™. I actually went out and installed Arch, so it may be some kind of inconsistency between distributions.

Reply:
No worries, this type of thing tends to happen with different environments :slightly_smiling_face:

Reply:
okay, I added it. Please tell me if I need to add any other headers